FSI/Displacement.C

```diff
@@ -41,9 +41,43 @@ void preciceAdapter::FSI::Displacement::write(double* buffer, bool meshConnectiv
     * the outer for the locations and the inner for the dimensions.
     * See the preCICE writeBlockVectorData() implementation.
     */
-    FatalErrorInFunction
-        << "Writing displacements is not supported."
-        << exit(FatalError);
+
+    // Copy the displacement field from OpenFOAM to the buffer
+
+    if (this->locationType_ == LocationType::faceCenters)
+    {
+        // For every boundary patch of the interface
+        for (const label patchID : patchIDs_)
+        {
+            // Write the displacement to the preCICE buffer
+            // For every cell of the patch
+            forAll(cellDisplacement_->boundaryField()[patchID], i)
+            {
+                for (unsigned int d = 0; d < dim; ++d)
+                    buffer[i * dim + d] =
+                        cellDisplacement_->boundaryField()[patchID][i][d];
+            }
+        }
+    }
+    else if (this->locationType_ == LocationType::faceNodes)
```

Comment:
Which of the two `locationType`s do you use for your setups. The `faceNodes` might run into an issue when used in parallel computations see https://github.com/precice/openfoam-adapter/issues/153

Reply:
In solids4foam, some of the solid solvers would prefer to use faceCenters and some would prefer to use faceNodes.

Thanks for making me aware of the #153. On the solids4foam side, there may be a workaround so that only faceCenters needs to be used.

Reply:
There are other workarounds available, e.g. https://github.com/precice/openfoam-adapter/issues/40#issuecomment-439922616 I guess being aware of the issue is just important when using it. Maybe adding a warning in the code linking to the issue would already be sufficient.

Reply:
OK, I added a warning in 3c2a9cc. Feel free to modify as you wish.